app/services/auto_assignment/assignment_service.rb

```diff
@@ -0,0 +1,99 @@
+class AutoAssignment::AssignmentService
+  pattr_initialize [:inbox!]
+
+  def perform_for_conversation(conversation)
+    return false unless assignable?(conversation)
+
+    agent = find_available_agent
+    return false unless agent
+
+    assign_conversation(conversation, agent)
+  end
```

Comment:
I don't see this method being used anywhere else, why do we need to keep it a public method?